developer/js/source/model-info-compiler/model-info-compiler.ts

```diff
@@ -0,0 +1,123 @@
+    *    "version": "1.0.0",
+    *    "languages": ["en"],
+    *    "authorName": "Example Author",
+    *    "authorEmail": "nobody@example.com",
+    *    "description": "Example wordlist model"
+    *  }
+    *
+    * For full documentation, see:
+    * https://help.keyman.com/developer/cloud/model_info/1.0/
+    */
+  let model_info: ModelInfoFile = JSON.parse(fs.readFileSync(sourceModelInfoFileName, 'utf8'));
+
+  //
+  // Build merged .model_info file
+  // https://api.keyman.com/schemas/model_info.source.json and
+  // https://api.keyman.com/schemas/model_info.distribution.json
+  // https://help.keyman.com/developer/cloud/model_info/1.0
+  //
+
+  function setModelMetadata(field: string, expected: any, warn: boolean = true) {
+    if(model_info[field] && model_info[field] !== expected) {
+      if(warn || typeof warn === 'undefined')
+        console.warn(`Warning: source ${sourceModelInfoFileName} field ${field} value "${model_info[field]}" does not match "${expected}" found in source file metadata.`);
+    }
+    model_info[field] = model_info[field] || expected;
+  }
+
+  //
+  // Merge model info file -- some fields have "special" behaviours -- see below
+  //
+
+  setModelMetadata('id', options.model_id);
+
+  setModelMetadata('name', options.kmpJsonData.info.name.description);
+  setModelMetadata('authorName', options.kmpJsonData.info.author.description);
+
+  // we strip the mailto: from the .kps file for the .model_info
+  setModelMetadata('authorEmail', options.kmpJsonData.info.author.url.match(/^(mailto\:)?(.+)$/)[2], false);
+
+  // extract the language identifiers from the language metadata
+  // arrays for each of the lexical models in the kmp.json file,
+  // and merge into a single array of identifiers in the
+  // .model_info file.
+
+  model_info.languages = model_info.languages || options.kmpJsonData.lexicalModels.reduce((a, e) => [].concat(a, e.languages.map((f) => f.id)), []);
+
+  setModelMetadata('lastModifiedDate', (new Date).toISOString());
+  setModelMetadata('packageFilename', path.basename(options.kmpFileName));
+
+  // Always overwrite with actual file size
+  model_info.packageFileSize = fs.statSync(options.kmpFileName).size;
+
+  setModelMetadata('jsFilename', path.basename(options.modelFileName));
+
+  // Always overwrite with actual file size
+  model_info.jsFileSize = fs.statSync(options.modelFileName).size;
+
+  // Always overwrite source data
+  model_info.packageIncludes = options.kmpJsonData.files.filter((e) => !!e.name.match(/.[ot]tf$/i)).length ? ['fonts'] : [];
+
+  setModelMetadata('version', options.kmpJsonData.info.version.description);
+
+  // The minimum Keyman version detected in the package file may be manually set higher by the developer
+  setModelMetadata('minKeymanVersion', minKeymanVersion, false);
+
+  if(options.sourcePath) {
+    setModelMetadata('sourcePath', options.sourcePath);
+  }
+
+  fs.writeFileSync(destModelInfoFileName, JSON.stringify(model_info, null, 2));
```

Comment:
Can you also pretty-print the kmp.json file generated in `kmp-compiler.ts` line 134?

Reply:
It would also impact `example.qaa.sencoten.model.kmp.json`